### ISSUE TITLE
Add lexicographical sorting of class level annotations

### ISSUE THREAD

**Issue by RutgerLubbers** [PicnicSupermarket/error-prone-support#1711]
### Problem

Method level annotations can be sorted lexicographically, but class level annotations cannot.

From `tech.picnic.errorprone.bugpatterns.LexicographicalAnnotationListing` there is a "todo" or "hint":
```
// XXX: Currently this checker only flags method-level annotations. It should likely also flag
// type-, field- and parameter-level annotations.
```

### Description of the proposed new feature

- [X] Support a stylistic preference.
- [-] Avoid a common gotcha, or potential problem.
- [-] Improve performance.

I want to create a new class to add sorting of class level annotations, say `LexicographicalListingOfClassLevelAnnotations`. This should flag and also rewrite

```java
@JBossLog
@AutoService(SomeClass.class)
public class SomeClass
```

into 

```java
@AutoService(SomeClass.class)
@JBossLog
public class SomeClass
```

### Considerations

This would still not sort annotations on members, nor on parameters.
I would want this to be a separate class, so one can include or exclude it if required.
Same for member-level an parameter-level annotations.

### Participation

- [X] I am willing to submit a pull request to implement this improvement.

How to go from here? I can create this code, add a test case.

### Additional context
<none>

**Comment by Stephan202** [comment]
Thanks for raising this issue @RutgerLubbers! Support for this is certainly welcome, though we would want to add this to the same check, rather than creating a new check. If users _really_ wish to constrain the tree type to which the check should apply, we could introduce a flag. But unless there's a compelling reason, I'd be in favour of omitting that complexity as well. As you might have noticed, this library is highly opinionated. 😉

As for what would then have to happen, at first glance:
1. Have `LexicographicalAnnotationListing` also implement `ClassTreeMatcher`.
2. Extract a `private match(Tree tree, ModifiersTree modifiers, VisitorState state)` method to which `matchClass and `matchMethod` then delegate.
3. Update `LexicographicalAnnotationListingTest` to cover the new logic. (No need to go overboard here; feel free to make a proposal, and we might push an additional commit if desired.)

If this sounds acceptable to you, we'd certainly accept a PR!

**Comment by RutgerLubbers** [comment]
Ok, I will do that.
